daemon/algod/server.go

```diff
@@ -84,6 +88,34 @@ func (s *Server) Initialize(cfg config.Local, phonebookAddresses []string, genes
 	s.log.SetLevel(logging.Level(cfg.BaseLoggerDebugLevel))
 	setupDeadlockLogger()
 
+	// Check some config parameters.
+	if cfg.RestConnectionsSoftLimit > cfg.RestConnectionsHardLimit {
+		s.log.Warnf(
+			"RestConnectionsSoftLimit %d exceeds RestConnectionsHardLimit %d",
+			cfg.RestConnectionsSoftLimit, cfg.RestConnectionsHardLimit)
+		cfg.RestConnectionsSoftLimit = cfg.RestConnectionsHardLimit
+	}
+	if cfg.IncomingConnectionsLimit < 0 {
+		return fmt.Errorf(
+			"Initialize() IncomingConnectionsLimit %d must be non-negative",
+			cfg.IncomingConnectionsLimit)
+	}
+
+	// Set large enough soft file descriptors limit.
+	var ot basics.OverflowTracker
+	fdRequired := ot.Add(
+		cfg.ReservedFDs,
+		ot.Add(uint64(cfg.IncomingConnectionsLimit), cfg.RestConnectionsHardLimit))
+	if ot.Overflowed {
+		return errors.New(
+			"Initialize() overflowed when adding up ReservedFDs, IncomingConnectionsLimit " +
+				"RestConnectionsHardLimit; decrease them")
+	}
+	err = util.SetFdSoftLimit(fdRequired)
+	if err != nil {
+		return fmt.Errorf("Initialize() err: %w", err)
```

Comment:
could you please be more specific in this error ?
i.e. Server.Initialize() failed when calling util.SetFdSoftLimit(%d) with the following error : %w